pandas/core/indexes/interval.py

```diff
@@ -441,23 +442,26 @@ def is_monotonic(self):
         Return True if the IntervalIndex is monotonic increasing (only equal or
         increasing values), else False
         """
-        return self._multiindex.is_monotonic
+        return self.is_monotonic_increasing
 
     @cache_readonly
     def is_monotonic_increasing(self):
         """
         Return True if the IntervalIndex is monotonic increasing (only equal or
         increasing values), else False
         """
-        return self._multiindex.is_monotonic_increasing
+        values = [self.right, self.left]
+
+        sort_order = np.lexsort(values)
+        return is_monotonic(sort_order, False)[0]
 
     @cache_readonly
     def is_monotonic_decreasing(self):
         """
         Return True if the IntervalIndex is monotonic decreasing (only equal or
         decreasing values), else False
         """
-        return self._multiindex.is_monotonic_decreasing
+        return self[::-1].is_monotonic_increasing
```

Comment:
I would prefer checking monotonic_decreasing directly, so:

```
values = [self.right, self.left]
sort_order = np.lexsort(values)
return is_monotonic(sort_order, False)[1]
```

(But adding the properties to  ``IntervalTree`` is even better).

Reply:
I have tried in this way. But `is_monotonic(sort_order, False)[1]` returns `False` for stationary IntervalIndex, e.g. `IntervalIndex([(0, 1], (0, 1]],
              closed='right',
              dtype='interval[int64]')`. By definition, `is_monotonic_decreasing` should be `True` for equal values.

`is_monotonic(sort_order, False)[0]` returns `True` for the stationary IntervalIndex.

Reply:
You are right, as lexsort does treats equal values as increasing their index location, so is_monotonic(...)[1] will fail. 

It would still deliver  efficiency gains from calculating ``is_monotonic_increasing`` and ``unique`` simultaneous using ``is_monotonic`` in ``IntervalTree`` as described above. Would you be up for adding this? Else this PR is good IMO.

Reply:
When implementing `is_unique` in this way, I got a similar problem. That is, `is_monotonic(sort_order, False)[2]` returns `False` for an IntervalIndex with consecutively identity element, e.g. `IntervalIndex([(0, 1], (0, 1], (2, 3]])` .

`IntervalIndex.is_unique` is being handled in #25159